### PR TITLE
Add docs for filters, focus, tags, and randomization of specs

### PR DESCRIPTION
**Pull request by Fryguy**
Based on the changes in https://github.com/crystal-lang/crystal/pull/8125, https://github.com/crystal-lang/crystal/pull/8310, and https://github.com/crystal-lang/crystal/pull/8068

### CODE REVIEW

File: guides/testing.md [hunk 1]
```diff
@@ -108,18 +113,75 @@ end
...
+it "adds", focus: true do
+  (2 + 2).should_not eq(5)
+end
+```
+
+If any such thing is marked with `focus: true` then only those examples will run.
+
+## Tagging specs
+
+Tags allow you to group your specs, allowing you to run a subset of your specs.
```
Comment (straight-shoota) [inline guides/testing.md]:
```suggestion
Tags can be used to group specs, allowing to only run a subset of specs when providing a `--tag` argument to the spec runner (see [Using the compiler](../using_the_compiler/README.md)).
```

File: guides/testing.md [hunk 2]
```diff
@@ -108,18 +113,75 @@ end
 
 They return the rescued exception so it can be used for further expectations, for example to verify specific properties of the exception.
 
+## Focusing on a group of specs
+
+A `describe`, `context` or `it` can be marked with `focus: true`, like this:
```
Comment (straight-shoota) [inline guides/testing.md]:
```suggestion
`describe`, `context` and `it` blocks can be marked with `focus: true`, like this:
```

File: guides/testing.md [hunk 3]
```diff
@@ -108,18 +113,75 @@ end
...
+end
+```
+
+If any such thing is marked with `focus: true` then only those examples will run.
+
+## Tagging specs
+
+Tags allow you to group your specs, allowing you to run a subset of your specs.
+
+A `describe`, `context` or `it` can be tagged, like this:
```
Comment (straight-shoota) [inline guides/testing.md]:
```suggestion
`describe`, `context` and `it` blocks can be tagged, like this:
```

File: guides/testing.md [hunk 4]
```diff
@@ -108,18 +113,75 @@ end
...
+  sleep 60
+  true.should be(true)
+end
+
+it "is fast", tags: "fast" do
+  true.should be(true)
+end
+```
+
+Tagging a `describe` or `context` implies that all of the containing `it`s are also tagged.
```
Comment (straight-shoota) [inline guides/testing.md]:
```suggestion
Tagging an example group (`describe` or `context`) extends to all of the contained examples.
```

File: guides/testing.md [hunk 5]
```diff
@@ -108,18 +113,75 @@ end
...
+
+Multiple tags can be specified by giving an Enumerable, such as Array or Set.
```
Comment (straight-shoota) [inline guides/testing.md]:
```suggestion
Multiple tags can be specified by giving an [`Enumerable`](https://crystal-lang.org/api/Enumerable.html), such as [`Array`](https://crystal-lang.org/api/Array.html) or [`Set`](https://crystal-lang.org/api/Set.html).
```

File: guides/testing.md [hunk 6]
```diff
@@ -108,18 +113,75 @@ end
...
 
 You can turn off colors with the switch `--no-color`.
 
+### Randomizing order of specs
+
+Specs, by default, run in the order defined, but can be run in a random order by passing `--order random` to `crystal spec`.
+
+Specs run in random order will display a seed value upon completion. This seed value can be used to rerun the specs in that same order by passing the seed value to `--order`.
+
+### Filtering
```
Comment (straight-shoota) [inline guides/testing.md]: Could you please move these changed spec runner CLI options to `using_the_compiler/README.md`?

File: guides/testing.md (hunk 2 above)
Comment (Fryguy) [reply guides/testing.md]: Note that the original code has this same text as a comment so it should probably be updated there as well.

File: using_the_compiler/README.md [hunk 7]
```diff
@@ -307,13 +307,20 @@ Run `crystal spec --options` for available options.
...
 * `--fail-fast`: Abort the spec run on first failure.
-* `--location <file>:<line>`: Runs example(s) at `line` in `file` (multiple options allowed).
 * `--junit_output <output_dir>`: Generates JUnit XML output.
 
+The following options can be combined to filter the list of specs to run.
+
+* `--example <name>`: Runs examples whose full nested names include `name`.
+* `--line <line>`: Runs examples whose line matches `line`.
+* `--location <file>:<line>`: Runs example(s) at `line` in `file` (multiple options allowed).
+* `--tag <tag>`: Runs examples with the specified tag, or excludes examples by adding ~ before the tag (multiple options allowed).
```
Comment (straight-shoota) [inline using_the_compiler/README.md]:
```suggestion
* `--tag <tag>`: Runs examples with the specified tag, or excludes examples by adding `~` before the tag (multiple options allowed).
```

**Pull request merged**